Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -638,6 +674,35 @@ template<typename T> Connection::SendSyncResult Connection::sendSync(T&& message
     return replyDecoder;
 }
 
+template<typename T> Connection::SendSyncResult<T> Connection::sendSync(T&& message, uint64_t destinationID, Timeout timeout, OptionSet<SendSyncOption> sendSyncOptions)
```

Comment:
Maybe you can investigate if the above function could call this function, so that there's no duplicated code.

Reply:
This works, just a matter of moving out the tuple and the Decoder ptr.